src/commands/commit/promote.ts

```diff
@@ -2,38 +2,82 @@ import { Flags } from "@oclif/core";
 
   static flags = {
     to: Flags.string({
       summary:
-        "The destination environment to promote changes from the preceding environment.",
-      required: true,
+        "The destination environment to promote all changes from the preceding environment.",
     }),
     force: Flags.boolean({
       summary: "Remove the confirmation prompt.",
     }),
+    only: Flags.string({
+      summary: "The ID of the commit to promote to the subsequent environment",
+    }),
   };
 
   async run(): Promise<void> {
     const { flags } = this.props;
 
-    // Confirm first as we are about to promote changes to go live in the target
-    // environment, unless forced.
-    const prompt = `Promote all changes to \`${flags.to}\` environment?`;
-    const input = flags.force || (await promptToConfirm(prompt));
-    if (!input) return;
+    // Currently this command support two modes of operation based on the flags:
+    // * --to : Promotes all changes to the destination environment.
+    // * --only: Promotes one commit to the subsequent enviroment
+    // The absence or presence of both flags will result in an error.
+    if (flags.to && flags.only) {
+      throw new Error(
+        "The flags `--to` and `--only` cannot be used together.\n See more help with --help",
+      );
+    }
+
+    if (!flags.to && !flags.only) {
+      throw new Error(
+        "You must specify either the `--to` or `--only` flag.\n See more help with --help",
+      );
+    }
+
+    if (flags.to) {
+      // Confirm first as we are about to promote changes to go live in the target
+      // environment, unless forced.
+      const prompt = `Promote all changes to \`${flags.to}\` environment?`;
+      const input = flags.force || (await promptToConfirm(prompt));
+      if (!input) return;
+
+      await withSpinner<ApiV1.PromoteAllChangesResp>(() =>
+        this.apiV1.promoteAllChanges(this.props),
+      );
+
+      this.log(
+        `‣ Successfully promoted all changes to \`${flags.to}\` environment`,
+      );
+    } else {
+      const prompt = `Promote commit \`${flags.only}\` ?`;
+      const input = flags.force || (await promptToConfirm(prompt));
+      if (!input) return;
 
-    await withSpinner<ApiV1.PromoteAllChangesResp>(() =>
-      this.apiV1.promoteAllChanges(this.props),
-    );
+      const resp = await withSpinner<ApiV1.PromoteChangeResp>(() =>
+        this.apiV1.promoteChange(this.props),
+      );
+      if (!isSuccessResp(resp)) {
+        const message = formatErrorRespMessage(resp);
+        return this.error(new ApiError(message));
+      }
 
-    this.log(
-      `‣ Successfully promoted all changes to \`${flags.to}\` environment`,
-    );
+      const { commit } = resp.data;
+      this.log(
+        `‣ Successfully promoted commit \`${flags.only}\` into \`${commit?.environment}\` environment`,
+      );
+      this.log(`‣ New commit ID: \`${commit?.id}\``);
+    }
```

Comment:
This is a small formatting thing but can we match this message format to how we are doing for the workflow run command? https://github.com/knocklabs/knock-cli/blob/6ff133cd9723ebd296dc34d2a83762f935a11fc5/src/commands/workflow/run.ts

So something like:
```
this.log(
  `‣ Successfully promoted the commit \`${flags.only}\` into \`${commit!.environment}\` environment`,
);
this.log(indentString(`Commit id: ${commit!.id}`, 4));
```